doc/examples/mnist_mlp.cpp

```diff
@@ -0,0 +1,272 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <cstdio>
+#include <functional>
+#include <iostream>
+#include <list>
+#include <math.h>
+#include <memory>
+#include <random>
+#include <set>
+#include <stdexcept>
+#include <string>
+
+#include <ngraph/graph_util.hpp>
+#include <ngraph/ngraph.hpp>
+
+#include "mnist.hpp"
+#include "tensor_utils.hpp"
+
+using namespace ngraph;
+
+size_t accuracy_count(const std::shared_ptr<runtime::TensorView>& t_sm,
+                      const std::shared_ptr<runtime::TensorView>& t_Y)
+{
+    const Shape& sm_shape = t_sm->get_shape();
+    size_t batch_size = sm_shape.at(0);
+    size_t label_size = sm_shape.at(1);
+    const Shape& Y_shape = t_Y->get_shape();
+    if (Y_shape.size() != 1 || Y_shape.at(0) != batch_size)
+    {
+        throw std::invalid_argument(
+            "Y and softmax shapes are incomptible");
+    }
+    size_t sm_pos = 0;
+    size_t count = 0;
+    for (size_t i = 0; i < batch_size; ++i)
```

Comment:
The loop structure here is a bit subtle. Could we somehow cast the data to a 2D array?